avalon/pipeline.py

```diff
@@ -248,6 +248,28 @@ def process(self, session, **kwargs):
         pass
 
 
+class InventoryAction(object):
+    """A custom action for the scene inventory tool
+
+    If registered the action will be visible in the Right Mouse Button menu
+    under the submenu "Actions".
+
+    """
+
+    label = None
+    icon = None
+    color = None
+    order = 0
+
+    def is_compatible(self, container):
+        """Override function in a custom class"""
+        return True
+
+    def process(self, containers):
+        """Override function in a custom class"""
```

Comment:
Improve docstring so it's clear that it's up to the developer to "filter out" the non-compatible containers. The action will trigger on "all!" actions, even invalid ones.